### PR TITLE
feat: separate hints

### PR DESCRIPTION
**Pull request by vankop**
This PR contains a:

- [ ] **bugfix**
- [x] new **feature**
- [ ] **code refactor**
- [x] **test update** 
- [ ] **typo fix**
- [ ] **metadata update**

### Motivation / Use-Case

partial https://github.com/webpack/schema-utils/pull/61

- moved hints in separated util
- added not logic support in hints
- added test for hints

### Breaking Changes

No

@evilebottnawi we can merge this in `dev-1`, if you don't want to merge partial feature (not logic support)

### CODE REVIEW

**Comment by vankop** [comment]
`npm audit` is annoying, did not find a way to fix it..

**Comment by evilebottnawi** [comment]
@vankop just ignore it :+1:

File: test/__snapshots__/index.test.js.snap [hunk 1]
```diff
@@ -611,6 +611,16 @@ exports[`Validation should fail validation for empty object 1`] = `
...
 `;
 
+exports[`Validation should fail validation for empty string #1 1`] = `
+"Invalid configuration object. Object has been initialized using a configuration object that does not match the API schema.
+ - configuration.emptyString should not be string."
+`;
+
+exports[`Validation should fail validation for empty string #2 1`] = `
+"Invalid configuration object. Object has been initialized using a configuration object that does not match the API schema.
+ - configuration.emptyString2 should be shorter than 1 characters."
```
Comment (evilebottnawi) [inline test/__snapshots__/index.test.js.snap:621]: `character`?

Comment (vankop) [reply test/__snapshots__/index.test.js.snap]: good catch. updated

**Comment by evilebottnawi** [comment]
Thanks!

**Pull request merged**